ionc/ion_timestamp.c

```diff
@@ -1074,25 +1084,34 @@ iERR _ion_timestamp_equals_helper(const ION_TIMESTAMP *ptime1, const ION_TIMESTA
         if (HAS_TZ_OFFSET(ptime1) ^ HAS_TZ_OFFSET(ptime2)) {
             goto is_false;
         }
+        if (HAS_TZ_OFFSET(ptime1) && (ptime1->tz_offset != ptime2->tz_offset)) {
+            goto is_false;
+        }
+        if (precision1 >= ION_TS_YEAR) {
+            if (ptime1->year != ptime2->year) {
+                goto is_false;
+            }
+        }
+        else {
```

Comment:
Consider moving this above the `precision1 >= ION_TS_YEAR` check above.  I.e.:
```
if (precision1 < ION_TS_YEAR) {
    FAILWITHMSG(...);
}
if (precision1 >= ION_TS_YEAR) {
   ...
```

This groups all the `precision1 >= ION_TS_...` checks together, and reads more easily.